dotcom-rendering/src/web/server/index.ts

```diff
@@ -19,18 +20,23 @@ function enhancePinnedPost(format: CAPIFormat, block?: Block) {
 const enhanceCAPIType = (body: unknown): CAPIArticleType => {
 	const data = validateAsCAPIType(body);
 
+	const enhancedBlocks = enhanceBlocks(
+		data.blocks,
+		data.format,
+		data.promotedNewsletter,
+	);
+
 	const CAPIArticle: CAPIArticleType = {
 		...data,
-		blocks: enhanceBlocks(
-			data.blocks,
-			data.format,
-			data.promotedNewsletter,
-		),
+		blocks: enhancedBlocks,
 		pinnedPost: enhancePinnedPost(data.format, data.pinnedPost),
 		standfirst: enhanceStandfirst(data.standfirst),
 		commercialProperties: enhanceCommercialProperties(
 			data.commercialProperties,
 		),
+		tableOfContents: data.config.switches.tableOfContents
```

Comment:
Do we want to consider driving this from format as well here so we can limit it to just Analysis and Explainer or are we happy to open this up to all articles?

Reply:
I did initially start with adding this switch check inside the component itself, along with the check on `format.design` but then I realised I'd have to pass in a new prop to include the switch and then it seemed more sensible to just do the check externally.

The check for `format` though is something we do inside the enhancer functions in several other places so I left that there to keep things consistent.

I'm not strong on this to be honest so if you think it works better to have all the checks externally or all internally I'm open to this, what do you think?

Reply:
> or are we happy to open this up to all articles?

I should have said, we have a check inside the enhancer like this

```ts
	if (
		format.design !== 'ArticleDesign' ||
		hasInteractiveContentsElement(blocks)
	) {
		return undefined;
	}
```

So it's only for Analysis for now

Reply:
Oh, wait, that says Article, not analysis...

Fixing!